client/js/s3/request-signer.js

```diff
@@ -265,7 +268,9 @@ qq.s3.RequestSigner = function(o) {
             pendingSignatureData = pendingSignatures[id],
             promise = pendingSignatureData.promise,
             signatureConstructor = pendingSignatureData.signatureConstructor,
-            errorMessage, response;
+            oldCredentials = Object.assign({}, credentialsProvider.get()),
```

Comment:
I have changed this so now `Object.assign` is used if it is available, if not then object cloning is performed manually with hardcoded fields.

I figured it is good to use `Object.assign` if available because in the future if the credentials object gets more fields added then `Object.assign` will pick them up as opposed to manually copying the object which will require us to add the additional fields in the copying code as well.